### PR TITLE
Added hw 2.4 solution

### PR DESCRIPTION
**Pull request by ElvenNurse**

### CODE REVIEW

File: src/main/java/core/basesyntax/FieldXY.java [hunk 1]
```diff
@@ -13,7 +15,106 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        int x = toX - robot.getX();
+        int y = toY - robot.getY();
+        if (x != 0) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FieldXY.java]: What's this check for?

File: src/main/java/core/basesyntax/FieldXY.java [hunk 2]
```diff
@@ -13,7 +15,106 @@
...
+            if (x > 0) {
+                moveRight(robot, x);
+            } else {
+                moveLeft(robot, x);
+            }
+        }
+        if (y != 0) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/FieldXY.java]: same

File: src/main/java/core/basesyntax/FieldXY.java (hunk 1 above)
Comment (ElvenNurse) [reply src/main/java/core/basesyntax/FieldXY.java]: x is the difference between current robot coordinate and target coordinate. this check if we need to move by x ordinate.

Comment (Tarabarnik) [reply src/main/java/core/basesyntax/FieldXY.java]: if x is 0 your robot won't move anyway, as you have a `for` loop at move methods with `i < x` check. So this check is redundant and tests will pass without them anyway

**Pull request closed without merging**